src/market_state.py

```diff
@@ -0,0 +1,71 @@
+from typing import NamedTuple
+
+from solana.publickey import PublicKey
+
+
+class AccountFlags(NamedTuple):
+    initialized: bool
+    market: bool
+    open_orders: bool
+    request_queue: bool
+    event_queue: bool
+    bids: bool
+    asks: bool
+
+
+class MarketState(NamedTuple):
+    account_flags: AccountFlags
+    own_address: PublicKey
+    vault_signer_nonce: int
+    base_mint: PublicKey
+    quote_mint: PublicKey
+    base_vault: PublicKey
+    base_deposits_total: int
+    base_fees_accrued: int
+    quote_vault: PublicKey
+    quote_deposits_total: int
+    quote_fees_accrued: int
+    quote_dust_threshold: int
+    request_queue: PublicKey
+    event_queue: PublicKey
+    bids: PublicKey
+    asks: PublicKey
+    base_lot_size: int
+    quote_lot_size: int
+    fee_rate_bps: int
+
+
+def create_account_flags(con) -> AccountFlags:
+    return AccountFlags(
+        initialized=con.initialized,
+        market=con.market,
+        open_orders=con.open_orders,
+        request_queue=con.request_queue,
+        event_queue=con.event_queue,
+        bids=con.bids,
+        asks=con.asks,
+    )
+
+
+def create_market_state(con) -> MarketState:
```

Comment:
This function can just be `__init__(self, ...).

Reply:
bump

Reply:
It would be appropriate to make it `__init__` but right now since we have also included the `base_spl_token_decimals` and `quote_spl_token_decimals` in the state, it will require network call, so I don't think putting it at `__init__` is the most appropriate.

re: dataclass I feel `NamedTuple` is more appropriate due to its immutability.